components/learn/modules/ROOT/pages/write-contracts.adoc

```diff
@@ -0,0 +1,158 @@
+= Write Solidity Contracts
+
+Welcome to the exciting world of smart contract development! This guide will let you get started writing Solidity contracts by going over the following:
+
+ * <<setting-up-a-solidity-project, Setting up a Solidity Project>>
+ * <<compiling-solidity-source-code, Compiling Solidity Source Code>>
+ * <<multi-file-projects, Multi-File Projects>>
+ * <<using-openzeppelin-contracts, Using the OpenZeppelin Contracts library>>
+
+=== About Solidity
+
+We won't cover language concepts such as syntax or keywords in this guide. For that, you'll want to check out the following curated content, which feature great learning resources for both newcomers and experienced developers:
+
+ * For a general overview of how Ethereum and smart contracts work, the official website hosts a https://ethereum.org/learn/[Learn about Ethereum] section with lots of beginner-friendly content
+ * If you're new to the language, the https://solidity.readthedocs.io/en/latest/introduction-to-smart-contracts.html[official Solidity documentation] is a good resource to have handy. Take a look at their https://solidity.readthedocs.io/en/latest/security-considerations.html[security recommendations], which nicely go over the differences between blockchains and traditional software platforms
+ * These https://consensys.github.io/smart-contract-best-practices/[best practices] are quite extensive, and include both https://consensys.github.io/smart-contract-best-practices/recommendations/[proven patterns] to learn from and https://consensys.github.io/smart-contract-best-practices/known_attacks/[known pitfalls] to avoid
+ * The https://solidity-05.ethernaut.openzeppelin.com/[Ethernaut] web-based game will have you look for subtle vulnerabilities in smart contracts as you advance through levels of increasing difficulty
+
+With that out of the way, let's get started!
+
+[[setting-up-a-solidity-project]]
+== Setting up a Solidity Project
+
+The first step in any project is to create a `contracts` directory, which is where you will store your Solidity source files (`.sol`). This is equivalent to the `src` directory you may be familiar with from other languages.
+
+We can now write our first simple smart contract:
+
+```solidity
+pragma solidity ^0.5.0;
+
+contract SetAndGet {
+    uint256 private value;
+
+    // Emitted when the stored value changes
+    event ValueChanged(uint256 newValue);
+
+    // Stores a new value in the contract
+    function set(uint256 newValue) public {
+      value = newValue;
+
+      emit ValueChanged(newValue);
+    }
+
+    // Reads the last stored value
+    function get() public view returns (uint256) {
+      return value;
+    }
+}
+```
+
+We will store it in `contracts/StoreAndGet.sol`. Each `.sol` file should have the code for a single contract, and be named after it.
```

Comment:
Hah, I should have scrolled further down before commenting. Still, I'd include it before (not after) the code block.